sklearn/feature_extraction/text.py

```diff
@@ -1117,21 +1117,30 @@ def fit(self, X, y=None):
         X : sparse matrix, [n_samples, n_features]
             a matrix of term/token counts
         """
+        X = check_array(X, accept_sparse='csc')
```

Comment:
Why casting to "csc" in fit ? Typical input would be "csr", wouldn't it?

Reply:
The only place that `X` is used is in `_document_frequency` which expect a `csc` matrix.

Reply:
FYI: https://github.com/scikit-learn/scikit-learn/pull/10443/files#diff-904f5f849e36c648b8b05de140d7fad8L576

Reply:
Maybe I am missing something but `_document_frequency` works fine with CSR matrices https://github.com/scikit-learn/scikit-learn/blob/48f3303bfc0be26136b98e9aa95dc3b3f916daff/sklearn/feature_extraction/text.py#L571-L574

Reply:
Uhm good point